apps/core/settingmanager.cpp

```diff
@@ -90,6 +93,22 @@ void SettingManager::removeAllTabs()
     DBManager::instance()->removeAllTabs();
 }
 
+void SettingManager::calculateCacheSize(QJSValue callback)
+{
+    if (!callback.isNull() && !callback.isUndefined() && callback.isCallable()) {
+        m_calculateCacheSizeCb.reset(new QJSValue(callback));
+    }
+    SailfishOS::WebEngine::instance()->notifyObservers(QString("get-cache-size"), QVariant());
```

Comment:
This notify should be inside the if-block as reported cache size is only handled when there is a callback.